src/scanner.c

```diff
@@ -418,12 +451,22 @@ bool tree_sitter_vim_external_scanner_scan(void *payload, TSLexer *lexer,
   if (valid_symbols[SEP_FIRST] && iswpunct(lexer->lookahead)) {
     s->separator = lexer->lookahead;
     advance(lexer, false);
+    s->ignore_comments = true;
     lexer->result_symbol = SEP_FIRST;
     return true;
+  // Allowed symbols around pattern in syntax commands is more restrive
+  } else if (valid_symbols[SYN_PATTERN_SEP_FIRST]
+      && is_valid_syn_pattern_delimiter(lexer->lookahead)) {
+    s->separator = lexer->lookahead;
+    advance(lexer, false);
+    s->ignore_comments = true;
+    lexer->result_symbol = SYN_PATTERN_SEP_FIRST;
+    return true;
```

Comment:
How do these two branches differ ? Do you have a specific example that matches one of the two branches and not the other ?

Reply:
Effectively both branches do the same thing, except that the second (SYN_PATTERN_FIRST_SEP) will not match for characters like '|' or '{' which are matched by `iswpunct`, valid to surround pattern in the `global` command but not around patterns in `syntax`.
Characters  matched by `is_valid_pattern_delimiter` are those I found in the syntax file for vim script (I found them on my system, did't find it using Google so I don't have a link to share).

I didn't found a proper way to check this without checking a value twice.

Reply:
Are you sure that those two character are not valid for `syntax` ?
In the docs it is said : `if the pattern contains a double quote, you can use another character that is not used in the pattern`.

Reply:
I thought I tested to assert this but opening a file with syntax using those characters as delimiters seems to work on my laptop. Yesterday I was on my main computer maybe there is a difference in the config (doubt but better check). I will try this tonight and make the fix accordingly